### PR TITLE
Singlecell eda

### PR DESCRIPTION
**Pull request by vicramr**

### CODE REVIEW

**Comment by vicramr** [comment]
I haven't read over the functions entirely yet, but so far pretty much everything looks good! You're correctly using the packaging guidelines: putting code in the `R` directory, running `devtools::document()` and committing any new files/changes, etc. Thanks!

I'll try running your functions next.

File: fblib/R/cell_viz.R [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          title = element_text(size = 12, colour = "#2f2f63"),
+          axis.ticks = element_line(colour = "#6b3447"),
+          plot.caption = element_text(size = 8, colour = "#2f2f63"),
+          plot.subtitle = element_text(size = 10, colour = "#2f2f63"))
+  
+  
+  sce<- scater::calculateQCMetrics(sce,exprs_values = "counts",compact=TRUE)
+  
+  col_qc_all<- SingleCellExperiment::colData(sce)$scater_qc$all 
+  cell_mean<- col_qc_all$total_counts/nrow(counts(sce))
```
Comment (vicramr) [inline fblib/R/cell_viz.R]: It looks like you're referring to `SingleCellExperiment::counts` here, so you should specify that. Or if you're calling a different `counts` function, also specify that.

File: fblib/R/cell_viz.R [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          axis.title = element_text(size = 10, colour = "#2f2f63"),
+          legend.title = element_text(size = 8, colour = "#2f2f63"),
+          legend.text = element_text(size = 8, colour = "#6b3447"),
+          title = element_text(size = 12, colour = "#2f2f63"),
+          axis.ticks = element_line(colour = "#6b3447"),
+          plot.caption = element_text(size = 8, colour = "#2f2f63"),
+          plot.subtitle = element_text(size = 10, colour = "#2f2f63"))
+  
+  
+  sce<- scater::calculateQCMetrics(sce,exprs_values = "counts",compact=TRUE)
```
Comment (vicramr) [inline fblib/R/cell_viz.R]: Is there a reason that you decided to set `compact=TRUE` and then access the QC metrics via `colData(sce)$scater_qc`? All those metrics are also available as columns of `sce`.

File: fblib/R/cell_viz.R [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  
+  col_qc_all<- SingleCellExperiment::colData(sce)$scater_qc$all 
+  cell_mean<- col_qc_all$total_counts/nrow(counts(sce))
+  cell_sum<- col_qc_all$total_counts
+  zero_exprs<- apply(counts(sce), 2, function(x) nnzero(x))
```
Comment (vicramr) [inline fblib/R/cell_viz.R]: `nnzero` -> `Matrix::nnzero`

Comment (vicramr) [reply fblib/R/cell_viz.R]: Also, how does apply work? Does 2 mean that the row dimension goes away, or the column dimension?

File: fblib/R/cell_viz.R [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  cell_mean_nonzero<- col_qc_all$total_counts/zero_exprs
+  non_zero_exprs <- data.frame(number = nrow(counts(sce)) - zero_exprs)
+  stat.value<- c(cell_mean, cell_sum)
+  stat.value.nonzero<- c(cell_mean_nonzero, cell_sum)
+  name<-c(rep("cell_mean", length(cell_mean)), rep("cell_sum", length(cell_sum)))
```
Comment (vicramr) [inline fblib/R/cell_viz.R]: Can you change these names to two words? Like, `"cell_mean"` -> `"Cell Mean"`? It looks better on the plots.

File: fblib/R/cell_viz.R (hunk 1 above)
Comment (aliviawu) [reply fblib/R/cell_viz.R]: Does it means I need to state what `counts(sce)` represents here or what the function `counts` does here?

File: fblib/R/cell_viz.R (hunk 2 above)
Comment (aliviawu) [reply fblib/R/cell_viz.R]: If compact=TRUE, the QC metrics are stored in the"scater_qc"field of thecolDataandrowDataas a nested DataFrame.  This avoids cluttering the metadata with QC metrics, especially if manyresults are to be stored in a single SingleCellExperiment object. 
So I extracted the return value from QC metrics using `colData(sce)$scater_qc`.

File: fblib/R/cell_viz.R (hunk 3 above)
Comment (aliviawu) [reply fblib/R/cell_viz.R]: `apply(...,2,...)` means the `function` is  applied to each column.

File: fblib/R/cell_viz.R (hunk 4 above)
Comment (aliviawu) [reply fblib/R/cell_viz.R]: OK.

File: fblib/R/cell_viz.R (hunk 1 above)
Comment (vicramr) [reply fblib/R/cell_viz.R]: Just prepend `packagename::` to it

File: fblib/R/cell_viz.R (hunk 2 above)
Comment (vicramr) [reply fblib/R/cell_viz.R]: I guess that's fine, but you should put some comments explaining that.

**Pull request merged**